src/mongoose/buildQuery.ts

```diff
@@ -307,22 +312,26 @@ class ParamParser {
             ],
           };
 
-          if (typeof formattedValue === 'number' || (typeof formattedValue === 'string' && mongoose.Types.ObjectId.isValid(formattedValue))) {
+          if (typeof formattedValue === 'string' && mongoose.Types.ObjectId.isValid(formattedValue)) {
             query.$or.push({
               [path]: {
                 [operatorKey]: formattedValue.toString(),
               },
             });
+
+            query.$or.push({
+              [path]: {
+                [operatorKey]: new mongoose.Types.ObjectId(formattedValue),
+              },
+            });
           }
 
-          if (typeof formattedValue === 'string') {
-            if (!Number.isNaN(formattedValue)) {
-              query.$or.push({
-                [path]: {
-                  [operatorKey]: parseFloat(formattedValue),
-                },
-              });
-            }
+          if (typeof formattedValue === 'string' && !Number.isNaN(formattedValue)) {
```

Comment:
What if formattedValue is a number? Also Number.isNaN does not work on strings (always false)

```suggestion
          if ((typeof formattedValue === 'string' || typeof formattedValue === 'number') && !Number.isNaN(Number(formattedValue))) {
```